### ISSUE TITLE
hypothesis.extra.pandas.column raises reportCallIssue

### ISSUE THREAD

**Issue by wangkev** [microsoft/pyright#7367]
**Describe the bug**
`hypothesis.extra.pandas.column` reports `error: No parameter named "name" (reportCallIssue)`.

**Code or Screenshots**
```python
from hypothesis.extra.pandas import column

foo = column(name="bar")
```

**VS Code extension or command-line**
Running as command line tool.

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

It appears that the hypothesis library is using the `attrs` library. The `attrs` library relies on the `dataclass_transform` mechanism introduced in PEP 681. The hypothesis library is using `attrs` in a way that is not supported by PEP 681, so it won't work with standards-compliant type checkers like pyright. The maintainers of hypothesis will need to make changes. Specifically, they'll need to add type annotations to the field definitions in their classes. It should look like this:

```python
@attr.s(slots=True)
class column:
    ...
    name: str | None = attr.ib(default=None)
```
